client/payment-details/dispute-details/dispute-awaiting-response-details.tsx

```diff
@@ -61,6 +88,156 @@ const DisputeAwaitingResponseDetails: React.FC< Props > = ( { dispute } ) => {
 								/>
 							</>
 						) }
+
+					{ /* Dispute Actions */ }
+					{ showDisputeActions && (
+						<Flex justify="start">
+							<Link
+								href={
+									// Prevent the user navigating to the challenge screen if the accept request is in progress.
+									isLoading
+										? ''
+										: getAdminUrl( {
+												page: 'wc-admin',
+												path:
+													'/payments/disputes/challenge',
+												id: dispute.id,
+										  } )
+								}
+							>
+								<Button
```

Comment:
Is there a reason for a button inside a link here? Would it be better to just have a button with a `href` attribute on it? 

There is the `wcTracks` event on the `<Button>` and the `href` on the `<Link>`

Can we combine these so there are two actions on the one element, rather than two actions over two elements?

Reply:
Ah, yeh this pattern came out of the discussion here: https://github.com/Automattic/woocommerce-payments/pull/7047#discussion_r1328174632.

We want the client-side routing AND native browser navigation via href when CMD+clicking. `Link` offers this, whereas `Button` doesn't – but we want the `Button` UI.

I'll consider if there's another way, perhaps there is a component that I don't know about.

Reply:
I've attempted a compromise by using a `Link` with a `span` child for the button styles.

```tsx
<Link
	href={ getAdminUrl( ... ) }
	onClick={ () => { wcpayTracks.recordEvent( ... ); } }
>
	<span className="components-button is-primary">
		I am styled like a Button
```

One risk with this approach is that we're using classNames for styling `components-button` detached from the `Button` component.

Reply:
> If I'm thinking about the project goals, getting this button right is critical. Using client-side routing is desirable, IMO, since the merchant will see the challenge form more quickly and smoothly → more challenges being initiated.

💯 totally agree. If we can speed up state transitions, that should improve "conversions" (responses to disputes), and of course make merchants' lives easier and more efficient. 

> One risk with this approach is that we're using classNames for styling components-button detached from the Button component.

I don't think using a raw `<span>` is much leaner than using a `<button>`. I'd lean towards using components as is (not borrowing classes or other hackery), and if there's a gap in the library, follow up by logging issues (and submitting PRs?) so the component library meets our needs. So personally I prefer the original `<Link><Button /></Link>

This reveals a potential UX/design issue – we might be using `Button` inappropriately when we should be using `Link`.

A bigger topic really, might be a good discussion on P2 e.g. codep2 or with other front end experts :)

Reply:
Thanks for your feedback @haszari and @brucealdridge.

Based on this discussion, I'm going to proceed with the `<Link><Button /></Link>` pattern to get this PR shipped.

I'll log a follow-up issue and move this conversation elsewhere.